h5py/_selector.pyx

```diff
@@ -179,6 +179,12 @@ cdef class Selector:
                     a = np.asarray(a)
                 if a.ndim != 1:
                     raise TypeError("Only 1D arrays allowed for fancy indexing")
+                # Fix for issue #1847, not sure this is optimal but it works
```

Comment:
I think this is probably fine. It's exactly what the code [was doing in h5py 2.x](https://github.com/h5py/h5py/blob/2.10.0/h5py/_hl/selections.py#L355-L358) before it broke, and I can't see an obvious better way.

Reply:
should I remove this line?

Reply:
I'm going to remove this.